app/main.py

```diff
@@ -1,3 +1,89 @@
-def shop_trip():
-    # write your code here
-    pass
+import json
+from datetime import datetime
+
+from app.customer import Customer
+from app.car import Car
+from app.shop import Shop
+from app.go_to_shop import count_trip_to_shop_value
+
+
+def shop_trip() -> None:
+
+    with open("app/config.json", "r") as data:
+        file_data = json.load(data)
+
+    customers = []
+
+    for customer in file_data["customers"]:
+        customers.append(Customer(
+            name=customer["name"],
+            product_cart=customer["product_cart"],
+            location=customer["location"],
+            money=customer["money"],
+            car=Car(
+                brand=customer["car"]["brand"],
+                fuel_consumption=customer["car"]["fuel_consumption"]
+            )
+        )
+        )
+
+    shops = []
+
+    for shop in file_data["shops"]:
+        shops.append(Shop(
+            name=shop["name"],
+            location=shop["location"],
+            products=shop["products"]
+        )
+        )
+
+    for customer in customers:
+        print(f"{customer.name} has {customer.money} dollars")
+        cheapest_shop, min_costs = count_trip_to_shop_value(
+            customer=customer,
+            shops=shops,
+            fuel_price=file_data["FUEL_PRICE"]
+        )
+
+        if min_costs > customer.money:
+            print(
+                f"{customer.name} doesn't have enough "
+                f"money to make purchase in any shop"
+            )
+            continue
+
+        print(f"{customer.name} rides to {cheapest_shop.name}\n")
+
+        customer_home = customer.location
+        customer.location = cheapest_shop.location
+
+        date = datetime(
+            year=2021, month=1, day=4, hour=12, minute=33, second=41
+        )
```

Comment:
Current datetime should be used

Reply:
not fixed

Reply:
tests crush if I make it  datetime.now().strftime('%d/%m/%Y %H:%M:%S')